content/articles/2018-11-tibble-2.0.0-pre-announce.Rmd

```diff
@@ -0,0 +1,222 @@
+
+```{r}
+library(tibble)
+
+(m <- cov(unname(iris[-5])))
+
+# problematic approach:
+# 1. make tibble
+# 2. apply nice names
+x <- as_tibble(m)
+colnames(x) <- letters[1:4]
+
+# better approach that works with tibble v1.4.2 AND dev tibble:
+# 1. apply nice names
+# 2. make tibble
+colnames(m) <- letters[1:4]
+as_tibble(m)
+
+```
+
+If possible, we recommend applying your "good" column names prior to calling `as_tibble()`. This creates code that works with tibble v1.4.2 and the development version, which is very appealing. For good examples, see these pull requests to [drake](https://github.com/ropensci/drake/pull/586), [prophet](https://github.com/facebook/prophet/pull/739), and [broom](https://github.com/tidymodels/broom/pull/536).
+
+It is also possible to use the new `.name_repair` argument in `tibble()` and `as_tibble()` (more below) to explicitly declare your intention around column names. This code would require `packageVersion("tibble") >= "2.0.0"`:
+
+```{r}
+# Alternative: use new `.name_repair` argument to permit dysfunctional names
+m <- cov(unname(iris[-5]))
+as_tibble(m, .name_repair = "minimal")
+
+# Alternative: use new `.name_repair` argument to fix dysfunctional names
+m <- cov(unname(iris[-5]))
+as_tibble(m, .name_repair = "unique")
+```
+
+What is the motivation for this increased attention to column names? The tibble package is offering stronger encouragement for names where each column can be identified by name and, preferably, without having to resort to backticks. Column names that don't meet these requirements are still allowed, but the user needs to permit them explicitly.
+
+After all, there are scenarios where problematic names should be tolerated. For example, after importing data, the user might need to inspect the data in order to determine which columns to keep. Or perhaps the column names contain data that is about to be converted to a proper variable with `gather()`.
+
+The `tibble()` constructor and the `as_tibble()` generic now support a new `.name_repair` argument that covers most use cases:
+
+- `"minimal"`: No name repair or checks, beyond basic existence.
+- `"unique"`: Make sure names are unique and not empty.
+- `"check_unique"`: (default value), no name repair, but check they are `unique`.
+- `"universal"`: Make the names `unique` and syntactic.
+- a function: apply custom name repair (e.g., `.name_repair = make.names` or `.name_repair = ~make.names(., unique = TRUE)` for names in the style of base R).
+
+See `` ?`name-repair` `` for more details.
+
+Packages that are in the business of making tibbles may even want to expose the `.name_repair` argument and pass it through to tibble. For example, this is the approach planned for [readxl](https://readxl.tidyverse.org), which reads rectangular data out of Excel workbooks.
```

Comment:
_"…pass it through to tibble"_ Do we mean `tibble()` here, or is this a reference to tibbles more broadly (i.e. both `tibble()` and `as_tibble()`)?